### PR TITLE
Unset `LLVM_ENABLE_RUNTIMES` in CMake caches

### PR DESCRIPTION
**Pull request by edymtt**
This would be needed for two reasons:
* avoid that testing for PRs that use `LLVM_ENABLE_RUNTIMES` to build compiler-rt may cause errors in unrelated PRs;
* avoid any fallout in CI should the above PRs be merged and subsequently reverted for unforeseen regressions.

Addresses rdar://100762636

### CODE REVIEW

File: utils/build-script-impl [hunk 1]
```diff
@@ -1831,6 +1831,7 @@ for host in "${ALL_HOSTS[@]}"; do
 
                 cmake_options+=(
                     -DLLVM_ENABLE_PROJECTS="$(join ";" ${llvm_enable_projects[@]})"
+                    -ULLVM_ENABLE_RUNTIMES
```
[the remark this answers is not in the document]
Comment (edymtt) [reply utils/build-script-impl:1842]: Good point -- I was rushing this after I realized my testing of #60993 may be causing failures in PR testing, and missed this.

Addressed in 10f581a85ba

**Review by compnerd — APPROVED** [review]
Assuming that the CI builders are happy, this seems safe-ish.

**Comment by edymtt** [comment]
@swift-ci please test

**Comment by edymtt** [comment]
@swift-ci please build toolchain macos

**Pull request merged**